src/main/java/com/minecolonies/commands/KillCitizen.java

```diff
@@ -100,6 +100,7 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
         sender.addChatMessage(new TextComponentString(String.format(COORDINATES_XYZ, position.getX(), position.getY(), position.getZ())));
         sender.addChatMessage(new TextComponentString(REMOVED_MESSAGE));
 
+        //todo: make this threadsafe
```

Comment:
onDeath actually kills it. Respawn is using setDead. And I am pretty sure that onDeath has to be called on the server thread...